src/rng.rs

```diff
@@ -1,80 +1,778 @@
     }
 
     #[inline]
     fn fill_bytes(&mut self, dest: &mut [u8]) {
-        self.r.fill_bytes(dest);
+        dest.copy_from_slice(self.0.squeeze_bytes(dest.len()).as_slice());
     }
 
     #[inline]
     fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), ark_std::rand::Error> {
-        Ok(self.r.fill_bytes(dest))
+        Ok(self.fill_bytes(dest))
     }
 }
 
-impl<D: Digest, R: RngCore + SeedableRng> FiatShamirRng for SimpleHashFiatShamirRng<D, R>
-where
-    R::Seed: From<[u8; 32]>,
-{
-    /// Create a new `Self` by initializing with a fresh seed.
-    /// `self.seed = H(initial_input)`.
-    #[inline]
-    fn initialize<'a, T: 'a + ToBytes>(initial_input: &'a T) -> Self {
-        let mut bytes = Vec::new();
-        initial_input
-            .write(&mut bytes)
-            .expect("failed to convert to bytes");
-        let seed = FromBytes::read(D::digest(&bytes).as_ref()).expect("failed to get [u8; 32]");
-        let r = R::from_seed(<R::Seed>::from(seed));
-        Self {
-            r,
-            seed: seed,
-            digest: PhantomData,
-        }
+impl<F: PrimeField> CryptographicSponge for SimplePoseidonRng<F> {
+    type Config = PoseidonConfig<F>;
+
+    fn new(params: &Self::Config) -> Self {
+        Self(PoseidonSponge::new(params))
     }
 
-    /// Refresh `self.seed` with new material. Achieved by setting
-    /// `self.seed = H(new_input || self.seed)`.
-    #[inline]
-    fn absorb<'a, T: 'a + ToBytes>(&mut self, new_input: &'a T) {
-        let mut bytes = Vec::new();
-        new_input
-            .write(&mut bytes)
-            .expect("failed to convert to bytes");
-        bytes.extend_from_slice(&self.seed);
-        self.seed = FromBytes::read(D::digest(&bytes).as_ref()).expect("failed to get [u8; 32]");
-        self.r = R::from_seed(<R::Seed>::from(self.seed));
+    fn absorb(&mut self, input: &impl Absorb) {
+        self.0.absorb(input);
+    }
+
+    fn squeeze_bytes(&mut self, num_bytes: usize) -> Vec<u8> {
+        self.0.squeeze_bytes(num_bytes)
+    }
+
+    fn squeeze_bits(&mut self, num_bits: usize) -> Vec<bool> {
+        self.0.squeeze_bits(num_bits)
+    }
+}
+
+/// Instantiate Poseidon sponge with default parameters
+impl<F: PrimeField> Default for SimplePoseidonRng<F> {
+    fn default() -> Self {
+        Self(PoseidonSponge::new(&poseidon_parameters_for_test()))
+    }
+}
+
+/// Mock trait for use in Marlin prover
+pub trait DefaultSpongeRNG: Default + CryptographicSponge + RngCore {}
+
+impl<F: PrimeField> DefaultSpongeRNG for SimplePoseidonRng<F> {}
+
+/// Generate default parameters (bls381-fr-only) for alpha = 17, state-size = 8
+fn poseidon_parameters_for_test<F: PrimeField>() -> PoseidonConfig<F> {
+    let alpha = 17;
+    let mds = vec![
+        vec![
+            F::from_str(
+                "43228725308391137369947362226390319299014033584574058394339561338097152657858",
+            )
+            .map_err(|_| ())
+            .unwrap(),
```

Comment:
You can use the [`MontFp`](https://docs.rs/ark-ff/latest/ark_ff/macro.MontFp.html) macro here

```suggestion
            ark_ff::MontFp!("43228725308391137369947362226390319299014033584574058394339561338097152657858"),
```